checker/src/main/java/org/checkerframework/checker/mustcall/MustCallVisitor.java

```diff
@@ -73,6 +75,35 @@ public Void visitReturn(ReturnTree tree, Void p) {
     return super.visitReturn(tree, p);
   }
 
+  @Override
+  public Void visitAssignment(AssignmentTree tree, Void p) {
+    // The following code implements the following rule:
+    //  * It is always safe to assign a MustCallAlias parameter of a constructor
+    //    to an owning field of the containing class.
+    // It is necessary to special case this because MustCallAlias is translated
+    // into @PolyMustCall.
+    ExpressionTree lhs = tree.getVariable();
+    ExpressionTree rhs = tree.getExpression();
+    Element lhsElt = TreeUtils.elementFromTree(lhs);
+    Element rhsElt = TreeUtils.elementFromTree(rhs);
+    if (lhsElt != null && rhsElt != null) {
+      boolean lhsIsOwningField =
+          lhs.getKind() == Tree.Kind.MEMBER_SELECT
+              && atypeFactory.getDeclAnnotation(lhsElt, Owning.class) != null;
```

Comment:
Does it matter what the base expression is here?  What if it is not `this`?

Reply:
(The new test case is `checker/tests/resourceleak/MustCallAliasNotThis.java`)

Reply:
I think this is fine, and thanks for the comment.  I'm hoping we can do a deeper re-think of `@CreatesMustCallFor` and support for non-final `@Owning` fields at some point.  For now I think some confusion at corner cases is inevitable.